x/evm/statedb/statedb.go

```diff
@@ -464,3 +472,12 @@ func (s *StateDB) SetError(err error) {
 
 	s.sdkError = err
 }
+
+func (s *StateDB) IBCTransfer(goCtx context.Context, msg *ibctransfertypes.MsgTransfer) (*ibctransfertypes.MsgTransferResponse, error) {
```

Comment:
It feels a bit out of place for `IBCTransfer()` to be in the statedb, but does seem to make it easier to add access to the keepers without breaking more things in geth.

To me I think it makes a bit more sense to pass in additional keepers to geth `vm.NewEVM()` then provide the matching keeper interface in `contract.accessibleState`. Since it doesn't seem needed by StateDB or the Ethermint keepers? (pls correct me if wrong)

Just cuts out an extra step passing the keepers around like so:
`Kava -> Ethermint -> EVM -> Custom precompile`  
`Kava -> EVM -> Custom precompile`

Reply:
sounds reasonable, the only downside is changing `func NewEVM(...) *EVM` signature will be a bit painful, it's widely used across go-ethereum codebase (40 direct occurrences)
cc @nddeluca